stellar/basic-payment-app
# This PR implements the functionality to enable interactions with Stellar Asset Contracts (SACs) in the Basic Payment App. It modifies the `signPayment` function in the `walletStore` to handle both regular payments and SAC transfers.

## Changes
- Updated `signPayment` function to detect contract addresses
- Implemented SAC transfer using `Contract.call()` for contract addresses
- Maintained existing functionality for regular payments
- Added error handling for SAC transfers

## Testing
I've tested this implementation with the following scenarios:
- Regular payment to a Stellar account
- SAC transfer to a contract address
- Error handling for invalid addresses and insufficient funds

To test this implementation:
1. Add a contract address to the contacts list
2. Attempt to send a payment to this contract address
3. Verify that the transaction is created as an SAC transfer
4. Test regular payments to ensure they still work as expected

Additional Notes
- This implementation assumes that the `contacts.isContractAddress()` function correctly identifies contract addresses
- Further testing on the Stellar testnet is recommended to ensure full compatibility with live SACs

Related Issue
Closes #15  (Enable interactions with Stellar Asset Contracts)
